woudc_data_registry/processing.py

```diff
@@ -396,43 +345,634 @@ def process_data(self, infile, verify_only=False, bypass=False):
-            'name': self.data_record.instrument_name,
-            'model': self.data_record.instrument_model,
-            'serial': self.data_record.instrument_number,
-            'x': self.data_record.x,
-            'y': self.data_record.y,
-            'z': self.data_record.z
+            'name': name,
+            'model': model,
+            'serial': serial,
+            'station_id': station,
+            'dataset_id': dataset,
+            'x': location[0],
+            'y': location[1],
+            'z': location[2]
         }
-        return data
 
-    def new_serial(self, instrument_id, verify_only):
-        data = self.get_instrument_data(instrument_id)
+        if force:
+            LOGGER.debug('Force-adding instrument. Skipping name/model check')
+            permission = True
+        else:
+            fields = ['name', 'model', 'station_id', 'dataset_id']
+            case_insensitive = ['name', 'model']
+            response = self.registry.query_multiple_fields(Instrument, model,
+                                                           fields,
+                                                           case_insensitive)
+            if response:
+                model['name'] = response.name
+                model['model'] = response.model
+                self.extcsv.extcsv['INSTRUMENT']['Name'] = response.name
+                self.extcsv.extcsv['INSTRUMENT']['Model'] = response.model
+
+                LOGGER.debug('All other instrument data matches.')
+                permission = True
+            else:
+                permission = False
 
-        fields = ['name',
-                  'model',
-                  'station_id',
-                  'dataset_id']
-        results = self.registry.query_multiple_fields(Instrument, data,
-                                                      fields)
-        if results:
-            LOGGER.debug('All other instrument data matches.')
+        if permission:
             LOGGER.info('Adding instrument with new serial number...')
+
             if verify_only:
-                LOGGER.info('Verification mode detected. '
-                            'Instrument not added.')
+                msg = 'Verification mode detected. Instrument not added.'
+                LOGGER.info(msg)
             else:
-                instrument = Instrument(data)
+                instrument = Instrument(model)
                 self.registry.save(instrument)
                 LOGGER.info('Instrument successfully added.')
             return True
         else:
             return False
 
+    def check_project(self):
+        """
+        Validates the instance's Extended CSV source file's #CONTENT.Class,
+        and returns True if no errors are found.
+        """
+
+        project = self.extcsv.extcsv['CONTENT']['Class']
+
+        LOGGER.debug('Validating project {}'.format(project))
+        self.projects = self.registry.query_distinct(Project.identifier)
+
+        if not project:
```

Comment:
Should be error per https://github.com/woudc/woudc-data-registry/blob/master/woudc_data_registry/processing.py#L182